inc/Engine/CDN/Subscriber.php

```diff
@@ -241,6 +248,54 @@ public function maybe_replace_url( $url, array $zones = [ 'all' ] ) {
 		return $url;
 	}
 
+	/**
+	 * Add a preconnect tag for the CDN.
+	 *
+	 * @since 3.8.3
+	 *
+	 * @param array  $urls          The initial array of wp_resource_hint urls.
+	 * @param string $relation_type The relation type for the hint: eg., 'preconnect', 'prerender', etc.
+	 *
+	 * @return array The filtered urls.
+	 */
+	public function add_preconnect_cdn( array $urls, string $relation_type ): array {
+		if (
+			'preconnect' !== $relation_type
+			||
+			rocket_bypass()
+			||
+			! $this->is_allowed()
+			||
+			! $this->is_cdn_enabled()
+		) {
+			return $urls;
+		}
+
+		$cdn_urls = $this->cdn->get_cdn_urls( [ 'all', 'images', 'css_and_js', 'css', 'js' ] );
+
+		if ( empty( $cdn_urls ) ) {
+			return $urls;
+		}
+
+		foreach ( $cdn_urls as $cdn_url ) {
+			// Note: As of 22 Feb, 2021 we cannot add more than one instance of a domain url
+			// on the wp_resource_hint() hook -- wp_resource_hint() will
+			// only actually print the first one.
+			// Ideally, we want both because CSS resources will use the crossorigin version,
+			// But JS resources will not.
+			// Jonathan has submitted a ticket to change this behavior:
+			// @see https://core.trac.wordpress.org/ticket/52465
+			// Until then, we order these to prefer/print the non-crossorigin version.
+			$urls[] = [ 'href' => $cdn_url ];
+			$urls[] = [
+				'href'        => $cdn_url,
+				'crossorigin' => 'anonymous',
```

Comment:
@iCaspar Is there a typo here?  shall it be cross_origin?

Reply:
shall the URLs in return here change based on CDN reserved for what? currently, it`s always the same independent on reserve CDN for which option

Reply:
On the typo, no it is correct as it is. The intent is to print a link with `crossorigin="anonymous"`, so WordPress will be looking (eventually) for that exact key.

On the reason that as of now the `crossorigin` version of the link will not be printed is because of the way WordPress currently handles the printing of those preconnect links. We can only print 1 version. ☹️ @GeekPress has submitted a patch to WP Core to allow both, and if/when (🤞) that happens, we should start seeing both. But for now, printing just the first, without crossorigin, is the expected behavior. See [this discussion](https://github.com/wp-media/wp-rocket/issues/3423#issuecomment-755428707) for a little more background.